src/parse.rs

```diff
@@ -906,6 +906,41 @@ impl<'a> Argv<'a> {
         Ok(())
     }
 
+    fn err_unknown_flag(&self, atom: &Atom) -> Result<(), String> {
+        use std::usize::MAX;
+        let mut best: String = "".to_string();
+        let flag = atom.to_string();
+        let mut min = MAX;
+    
+        let mut possibles = Vec::new();
+
+        for (key, _) in self.dopt.descs.synonyms() {
+            possibles.push(key);
+        }
+
+        for key in self.dopt.descs.keys() {
+            possibles.push(key);
+        }
+
+        for key in possibles.iter() {
+            match *key {
+                &Short(_) | &Long(_) | &Command(_) => {
+                    let dist = levenshtein(&flag, &key.to_string());
+                    if dist < 3 && dist < min {
+                        min = dist;
+                        best = key.to_string()
```

Comment:
This allocation seems unnecessary. It's the exact same string as the one passed to `levenshtein`.